src/safeds/data/tabular/containers/_table.py

```diff
@@ -663,6 +706,30 @@ def add_rows(self, rows: list[Row] | Table) -> Table:
         result = pd.concat([result, *row_frames]).infer_objects()
         result.columns = self.column_names
         return Table._from_pandas_dataframe(result)
+        """
+        if isinstance(rows, Table):
+            rows = rows.to_rows()
+        int_columns = []
+        for row in rows:
+            if self.number_of_rows == 0:
+                int_columns = list(filter(lambda name: isinstance(row[name], int | np.int64), row.column_names))
+                if self.number_of_columns == 0:
+                    for column in row.column_names:
+                        self._data[column] = Column(column, [])
```

Comment:
```suggestion
                        self._data[column] = Column(column, [])
                elif self.column_names != row.column_names:
                    raise SchemaMismatchError
```

Reply:
Same question here: Why would you only check for column names if u can check the whole schema (as currently done)?